cmd/node/factory/structs.go

```diff
@@ -493,10 +506,24 @@ func ProcessComponentsFactory(args *processComponentsFactoryArgs) (*Process, err
 	}, nil
 }
 
+func setGenesisHeader(args *processComponentsFactoryArgs, genesisBlocks map[uint32]data.HeaderHandler) error {
+	genesisBlock, ok := genesisBlocks[args.shardCoordinator.SelfId()]
+	if !ok {
+		return errors.New("genesis block does not exist")
+	}
+
+	err := args.data.Blkc.SetGenesisHeader(genesisBlock)
+	if err != nil {
+		return err
+	}
+
```

Comment:
SetGenessisHeaderHash as well

Reply:
I have checked the code and It is not needed, only nonce and round is used in newValidatorStatisticsProcessor, until the call of prepareGenesisBlock which will set again the genesis header, genesis hash with the correct validator stats root hash inside, and put them in the storage.